core/src/main/java/org/apache/accumulo/core/clientImpl/mapreduce/lib/ConfiguratorBase.java

```diff
@@ -189,7 +192,7 @@ public static void setConnectorInfo(Class<?> implementingClass, Configuration co
     checkArgument(tokenFile != null, "tokenFile is null");
 
     try {
-      DistributedCacheHelper.addCacheFile(new URI(tokenFile), conf);
+      DistributedCacheHelper.addCacheFile(new URI(tokenFile + "#" + tokenCacheMnemonic), conf);
```

Comment:
If the names are confusing, you could just rename `tokenFile` to `tokenFilePath`.  Just make sure to update the javadoc